portal-impl/src/com/liferay/portal/security/permission/StagingPermissionChecker.java

```diff
@@ -134,8 +178,17 @@ public boolean hasPermission(
 			return true;
 		}
 
-		return _permissionChecker.hasPermission(
-			liveGroup, name, primKey, actionId);
+		if (group != null) {
+			setGroupId(group.getGroupId());
```

Comment:
Use CentralizedThreadLocal.setWithSafeClosable() and remove the removeGroupId() method.

Reply:
ok!